### ISSUE TITLE
Active Task not easily distinguishable

### ISSUE THREAD

**Issue by fastrizwaan** [micheleg/dash-to-dock#434]
Hi,

Both Active and running applications show same color of Dots. Also the lighter box around the icon for current active window, is too subtle and not easily noticeable. Please see the screenshot, the dash-to-dock settings in Dash-to-Dock (lighter background) and in Dash-to-Panel (highlighted underline).

We should have something more prominent so that active application can be noticed easily visually. Thanks.

![screenshot from 2017-01-14 16-06-14](https://cloud.githubusercontent.com/assets/1161682/21954350/2252f5c2-da74-11e6-8c99-f4f1ce13f718.png)

**Comment by fastrizwaan** [comment]
If we enable custom themes, then the highlighter is almost invisible. Please see that nautilus is currently active, and if we look at dash it is difficult to distinguish and on panel it is very clear with the underline.
![screenshot from 2017-01-14 16-18-20](https://cloud.githubusercontent.com/assets/1161682/21954392/4eebc158-da75-11e6-8a29-c66b81b0de09.png)

**Comment by fastrizwaan** [comment]
Could we use Underlines like gtk3 tabs on left side?

**Comment by micheleg** [comment]
If we use underline, don't we lose the windows counter indicators? I agree someone might like a more prominent indicator for the active application, but this doesn't look like a good solution. 

What do you mean by "If we enable custom themes" ?

**Comment by fastrizwaan** [comment]
I use zuki theme, which is a custom theme, and the highlight is even less prominent.

A dash followed by dots

We could have dash+dots together in this manner
_...  = first window active rest are minimized
._.. = second window active; 1st and others are minimized/inactive
_._ = 1st and 3rd window active 2nd is inactive/minimized

or if we don't want to go too deep, then active dash with dots in the center, could show both multiple windows and active window.

__.__ = 1 active 1 inactive
__..__ = 1 active 2 inactive
__...__ = 1 active 3 or more inactive

Thanks.

**Comment by Baltix** [comment]
@fastrizwaan it seems you are suggesting Morse code ;)

**Comment by micheleg** [comment]
The first idea seems a bit overkill to me, while I need to see the second visually.

From an implementation point of view, it could be done changing the function drawing the circles  [https://github.com/micheleg/dash-to-dock/blob/master/appIcons.js#L374-#L447]. I think CSS is too limited for this (unless we just want a line in the background).

**Comment by redvimo** [comment]
I have an idea, not much thought went into it though..
Instead of indicating a running application with underlines or dots. What if the running app was indicated by a background (behind the icon), like when you hover over an icon it shows that grey block. But instead of it being grey, it should be that nice gnome blue color that's being used for the underline. Then for the hover indicator it should be the same blue but with alot more transparency and it must be on top of the icon not behind. Then to show how many windows or instances are open for that running app, there should be a small block with a number in the bottom right hand corner.
I will post pictures soon of my idea.

**Comment by fastrizwaan** [comment]
dash to panel which brings the windows 10 task bar feature to gnome desktop has implemented various multi-window active and inactive indicators, including the show desktop vertical bar at the end of the panel. 

dash to dock has different approach and work-flow, and dash-to-dock requires somewhat similar indicators for active and inactive multi windows status on the dock.

dash to panel seems to have gotten the features from https://github.com/zagortenay333/ciliora-secunda-shell 

please see screenshot: [https://camo.githubusercontent.com/8b1c0f2aea1130f5dcdafe7bd759b060699a4092/687474703a2f2f6f72696731322e64657669616e746172742e6e65742f613061392f662f323031362f3330302f622f332f63696c696f72615f736563756e64615f7368656c6c5f62795f7a61676f7274656e61793333332d643836717a6d322e706e67](url)

Thanks.

**Comment by micheleg** [comment]
@redvimo I'm not sure I understand your proposed design, I'm waiting for the picture. 

@fastrizwaan: it could be nice to port at least some of the options (but not all of them!). Not my main priority though.

**Comment by redvimo** [comment]
@michel Ill draw up something with inkscape. Would that be ok?

**Comment by redvimo** [comment]
@micheleg Here is the picture: (The Icon with blue background is for currently focused application and the grey background is for all other applications that are running and hovering over with the mouse)
![dash-to-dock-enhancement](https://cloud.githubusercontent.com/assets/16490763/24077082/f37e2b88-0c49-11e7-8237-9ccab87879ac.png)

**Comment by micheleg** [comment]
@redvimo  I agree the current style for the active application is not very distinguishable, although I personally don't feel the need to get such a strong indication. Regarding the numbers instead, I''m not much convinced it is a good idea. Again, there's no real need to know exactly how many windows you have in most occasions. Moreover, such a counter reminds a bit of a notification indicator as found in other platforms, and might be quite a bit confusing. I'm will consider these suggestions if I come to improve and implement additional launcher indicator styles.

**Comment by redvimo** [comment]
@micheleg Yeah i guess u right. Its not an issue for me, im quite happy
with the way it is. I was just throwing some ideas around.
Thanks for the consideration.

**Comment by franglais125** [comment]
At least for the focus style, I always increase the alpha value on stylesheet.css, as (IMO) it's barely noticeable by default.

@micheleg what styles would you prefer be imported from dash-to-panel? It'd be nice to have a few of them for d-t-d as well, although as you say perhaps with not that many options.

**Comment by weltenweber** [comment]
Are there any updates on this? The single thing I’m missing from Dash-To-Panel (which is unusable on small screens because of no option to enable intellihide) are the more flexible and discernable styles for running and focussed applications.

**Comment by micheleg** [comment]
The styles of the indicators (dots, squares, dashes etc.) have been ported already and are available since version 62. The option so assign different colous to the active task has not been ported instead. It should not be difficult to port that also, just nobody invested time into it. We did try to make the active task a but more prominent quite some time ago [68a6ede76dee8f3]. Also note that the styling should be customizable at the theme level through the ```.focused``` css style class.
